functions/taskrouterListeners/transfersListener.private.ts

```diff
@@ -0,0 +1,225 @@
+  taskAttributes.transferMeta;
+
+/**
+ * Checks the event type to determine if the listener should handle the event or not.
+ * If it returns true, the taskrouter will invoke this listener.
+ */
+export const shouldHandle = (event: EventFields) => eventTypes.includes(event.EventType);
+
+export const handleEvent = async (context: Context<EnvVars>, event: EventFields) => {
+  try {
+    if (!shouldHandle(event)) return;
+
+    const {
+      EventType: eventType,
+      TaskChannelUniqueName: taskChannelUniqueName,
+      TaskSid: taskSid,
+    } = event;
+
+    console.log(`===== Executing TransfersListener for event: ${eventType} =====`);
+
+    const taskAttributes = JSON.parse(event.TaskAttributes!);
+
+    /**
+     * If a chat transfer gets accepted, it should:
+     * 1) Complete the original task
+     */
+    if (isChatTransferAccepted(eventType, taskChannelUniqueName, taskAttributes)) {
+      console.log('Handling chat transfer accepted...');
+
+      const { originalTask: originalTaskSid } = taskAttributes.transferMeta;
+      const client = context.getTwilioClient();
+
+      await client.taskrouter
+        .workspaces(context.TWILIO_WORKSPACE_SID)
+        .tasks(originalTaskSid)
+        .update({
+          assignmentStatus: 'completed',
+          reason: 'task transferred accepted',
+        });
+
+      console.log('Finished handling chat transfer accepted.');
+    }
+
+    /**
+     * If a chat transfer gets rejected, it should:
+     * 1) Adjust original task attributes:
+     *    - channelSid: from 'CH00000000000000000000000000000000' to original channelSid
+     *    - transferMeta.sidWithTaskControl: to original reservation
+     * 2) Cancel rejected task
+     */
+    if (isChatTransferRejected(eventType, taskChannelUniqueName, taskAttributes)) {
+      console.log('Handling chat transfer rejected...');
+
+      const { originalTask: originalTaskSid } = taskAttributes.transferMeta;
+      const client = context.getTwilioClient();
+
+      const originalTask = await client.taskrouter
+        .workspaces(context.TWILIO_WORKSPACE_SID)
+        .tasks(originalTaskSid)
+        .fetch();
+
+      const { attributes: attributesRaw } = originalTask;
+      const originalAttributes = JSON.parse(attributesRaw);
+
+      const { channelSid } = taskAttributes;
+      const attributesWithChannelSid = {
+        ...originalAttributes,
+        channelSid,
+        transferMeta: {
+          ...originalAttributes.transferMeta,
+          sidWithTaskControl: originalAttributes.transferMeta.originalReservation,
+        },
+      };
+
+      await client.taskrouter
+        .workspaces(context.TWILIO_WORKSPACE_SID)
+        .tasks(originalTaskSid)
+        .update({
+          attributes: JSON.stringify(attributesWithChannelSid),
+        });
+
+      await client.taskrouter.workspaces(context.TWILIO_WORKSPACE_SID).tasks(taskSid).update({
+        assignmentStatus: 'canceled',
+        reason: 'task transferred rejected',
+      });
```

Comment:
Does it makes sense to run this two in parallel?

Reply:
Nice idea!